src/github/client/key-locator.ts

```diff
@@ -7,7 +7,7 @@ import isBase64 from "is-base64";
 /**
  * Look for a Github app's private key
  */
-export const keyLocator = async (gitHubAppId?: number) => {
+export const keyLocator = async (gitHubAppId: number | undefined) => {
```

Comment:
in TS, `gitHubAppId?: number` is the same type as `gitHubAppId: number | undefined`, the `?` is just syntactic sugar and makes it more concise IMO.  The former is the standard we want to use to reduce boilerplate.

Reply:
So for example, in bellow screenshot, if I use `gitHubAppId: number | undefined`, it will prompt me a TS error
<img width="1243" alt="image" src="https://user-images.githubusercontent.com/105693507/185000694-a258b6ee-701e-4560-bcb2-6c98f1f9ba36.png">

But `gitHubAppId?: number` does not.
<img width="1221" alt="image" src="https://user-images.githubusercontent.com/105693507/185000730-7002d58a-64f1-44e3-b52f-ff20fd6b34d3.png">


So my goal is to ensure, our code doesn't forget to provide the `gitHubAppId`, even it is undefined. Otherwise, we maybe forgot to modify somewhere in our code to supply to `gitHubAppId` and still pass the TS checking, which will mix cloud data with GHE data.

Reply:
It is related to recent series of PR that try to fix the `gitHubInstallationId` is not unique in GHE, I am a bit concern somewhere in the code we forgot to add the `gitHubAppId`.

Reply:
Are you saying we should make gitHubAppId a required arg instead of optional? If that's the case, we would need to update every method in Subscription that has the gitHubAppId as a param and also update every call to each of those methods. From what I can see, getAllFiltered, getSingleInstallation, findOneForGitHubInstallationId, and getAllForInstallation would need to be updated and it would be best to break this out into 3 small prs to also address updating where these methods are being called:

- Subscription.getAllForInstallation is called in 7 places: should be it's own PR
- Subscription.getSingleInstallation is called in close to 20 places: should be it's own PR
- Subscription.getAllFiltered and Subscription.findOneForGitHubInstallationId are both only called in 3 places so could be tackled together.